frontend/src/concepts/k8s/K8sNameDescriptionField/utils.ts

```diff
@@ -0,0 +1,121 @@
+import * as _ from 'lodash-es';
+import {
+  getDescriptionFromK8sResource,
+  getDisplayNameFromK8sResource,
+  isK8sDSGResource,
+  isValidK8sName,
+  translateDisplayNameForK8s,
+} from '~/concepts/k8s/utils';
+import { RecursivePartial } from '~/typeHelpers';
+import {
+  K8sNameDescriptionFieldData,
+  K8sNameDescriptionFieldUpdateFunctionInternal,
+  UseK8sNameDescriptionDataConfiguration,
+} from './types';
+
+/**
+ * Those used in OpenShift Routes need to be less than 63 between namespace & resource name.
+ * 30 because UX feels this is an easier number to understand than 31 or 32.
+ * Arguably we could dynamically figure this out, but as of right now splitting equally is the UX.
+ */
+const ROUTE_BASED_NAME_LENGTH = 30;
+/**
+ * These are resources that need to be limited
+ * TODO: Support more than just ROUTE_BASED_NAME_LENGTH
+ */
+export enum LimitNameResourceType {
+  /** Project names take up half of all in-project route lengths */
+  PROJECT,
+  /** Workbenches create routes */
+  WORKBENCH,
+  // TODO: Support Model Serving?
+}
+/** K8s max DNS subdomain name length */
+const MAX_RESOURCE_NAME_LENGTH = 253;
+
+export const setupDefaults = ({
+  initialData,
+  limitNameResourceType,
+  safePrefix,
+}: UseK8sNameDescriptionDataConfiguration): K8sNameDescriptionFieldData => {
+  let initialName = '';
+  let initialDescription = '';
+  let initialK8sNameValue = '';
+  let configuredMaxLength = MAX_RESOURCE_NAME_LENGTH;
+
+  if (isK8sDSGResource(initialData)) {
+    initialName = getDisplayNameFromK8sResource(initialData);
+    initialDescription = getDescriptionFromK8sResource(initialData);
+    initialK8sNameValue = initialData.metadata.name;
+  }
+  if (limitNameResourceType != null) {
```

Comment:
As of right now -- it's not figuring out anything. There are comments along with the type & constant used inside this if-statement. I could make it more complex and use the mapping of the ENUM to get the value -- but the only use-case for limiting resource name based on type is the Route issue today.

I could clean this up if you feel this is uncomfortable. Since it's an enum -- TS safe-guards this from being "anything"

Reply:
just making sure its known to be working like this today.